code/modules/antagonists/voidwalker/voidwalker_kidnap.dm

```diff
@@ -0,0 +1,173 @@
+/// Component that lets us space kidnap people as the voidwalker with our HAAAADS
+/datum/component/space_kidnap
+	/// How long does it take to kidnap them?
+	var/kidnap_time = 6 SECONDS
+	/// Are we kidnapping right now?
+	var/kidnapping = FALSE
+
+/datum/component/space_kidnap/Initialize(...)
+	if(!ishuman(parent))
+		return COMPONENT_INCOMPATIBLE
+
+	RegisterSignal(parent, COMSIG_LIVING_UNARMED_ATTACK, PROC_REF(try_kidnap))
+
+/datum/component/space_kidnap/proc/try_kidnap(mob/living/parent, atom/target)
+	SIGNAL_HANDLER
+
+	if(!isliving(target))
+		return
+
+	var/mob/living/victim = target
+
+	if(victim.stat == DEAD)
+		target.balloon_alert(parent, "is dead!")
+		return COMPONENT_CANCEL_ATTACK_CHAIN
+
+	if(!victim.incapacitated())
+		return COMPONENT_CANCEL_ATTACK_CHAIN
+
+	if(!isspaceturf(get_turf(target)))
+		target.balloon_alert(parent, "not in space!")
+		return COMPONENT_CANCEL_ATTACK_CHAIN
+
+	if(!kidnapping)
+		INVOKE_ASYNC(src, PROC_REF(kidnap), parent, target)
+		return COMPONENT_CANCEL_ATTACK_CHAIN
+
+/datum/component/space_kidnap/proc/kidnap(mob/living/parent, mob/living/victim)
+	victim.Paralyze(kidnap_time) //so they don't get up if we already got em
+	var/obj/particles = new /obj/effect/abstract/particle_holder (victim, /particles/void_kidnap)
+	kidnapping = TRUE
+
+	if(do_after(parent, kidnap_time, victim, extra_checks = CALLBACK(victim, TYPE_PROC_REF(/mob, incapacitated))))
+		take_them(victim)
+
+	qdel(particles)
+	kidnapping = FALSE
+
+/datum/component/space_kidnap/proc/take_them(mob/living/victim)
+	if(ishuman(victim))
+		var/mob/living/carbon/human/hewmon = victim
+		hewmon.gain_trauma(/datum/brain_trauma/voided)
+
+	victim.flash_act(INFINITY, override_blindness_check = TRUE, visual = TRUE, type = /atom/movable/screen/fullscreen/flash/black)
+	new /obj/effect/temp_visual/circle_wave/unsettle(get_turf(victim))
+
+	if(!SSmapping.lazy_load_template(LAZY_TEMPLATE_KEY_VOIDWALKER_VOID) || !GLOB.voidwalker_void.len)
+		dump(victim)
+		victim.heal_overall_damage(brute = 80, burn = 20)
+		CRASH("[victim] was instantly dumped after being voidwalker kidnapped due to a missing landmark!")
+	else
+		victim.heal_and_revive(90)
+
+		victim.adjustOxyLoss(-100, FALSE)
+
+		new /obj/effect/wisp_mobile (get_turf(pick(GLOB.voidwalker_void)), victim)
+		addtimer(CALLBACK(src, PROC_REF(dump), victim), 60 SECONDS)
+
+/datum/component/space_kidnap/proc/dump(mob/living/trash)
```

Comment:
If this proc is on the component, and the component is on the voidwalker, does gibbing the voidwalker while someone is kidnapped trap the victim in space baby jail?
Maybe give them a status effect to handle this instead